runners/google-cloud-dataflow-java/worker/src/main/java/org/apache/beam/runners/dataflow/worker/StreamingDataflowWorker.java

```diff
@@ -2377,12 +2399,48 @@ public List<Windmill.KeyedGetDataRequest> getKeysToRefresh(Instant refreshDeadli
           ShardedKey shardedKey = entry.getKey();
           for (Work work : entry.getValue()) {
             if (work.getStartTime().isBefore(refreshDeadline)) {
-              result.add(
+              Map<Windmill.LatencyAttribution.State, Duration> durations =
```

Comment:
it would be simpler to just build this map up in the work item instead of one keyed by State
That saves merging here since it's already handled there. just need translation between enums

Reply:
Done (after next force push)